libpod/reset.go

```diff
@@ -150,5 +151,25 @@ func (r *Runtime) Reset(ctx context.Context) error {
 		prevError = err
 	}
 
+	path, err := exec.LookPath("systemctl")
+	if err != nil {
+		logrus.Info("systemctl not found, podman service will not be restarted")
+	} else {
+		cmd := exec.Command(path)
+		if rootless.IsRootless() {
+			cmd.Args = append(cmd.Args, []string{"--user", "restart", "podman.service"}...)
```

Comment:
I think this should be try-restart.  We only want services started if they are already running. 
But I am not sure this is relevant, since podman.service defaults to socket activation. We probably should shut the service down when the system is reset and then rely on socket activation to start it.

Although with try-restart, this should be fine, and would fix a race condition, where the service is kicked off after it has been shutdown.

Reply:
I was debating between try-restart and restart. I will convert to that and adjust the test accordingly

Reply:
Are we doing systemctl service stop before the reset as well?

Reply:
@rhatdan  so, should we plan on stopping and restarting since try-restart only restarts running services or should we just stop and let it be started later?

Reply:
I opted for the `systemctl --user stop podman.service` which leads service activation to be re initiated by anything that uses `podman.socket`